### DISCUSSION TITLE
Accessing Vee-Validate validation object

### DISCUSSION THREAD

**Discussion by jhrr** [formvuelate/formvuelate#276] [General]
Hi there, 

Maybe I'm overlooking something obvious here, but I'm using Formvuelate with the Vee-Validate plugin and I'd like to get hold of the Vee-Validate, form-level, `validation` state object outside of the template (ie. in my `setup`) so I can use it in my application logic. Is this at all possible?

Thanks for all the effort on the project.

**Comment by logaretm** [comment]
Hello, at the moment this is not possible because of the way plugins work at the moment in FVL. To elaborate, the plugins compose their "setup" function into the `SchemaForm` component's setup, this makes them inaccessible from your component which is usually the `SchemaForm` parent.

We discussed this in https://github.com/formvuelate/formvuelate/discussions/204

This will be taken into consideration in v4 which will be a re-write for most plugin APIs and other parts of FVL.

**Reply by jhrr** [reply]
Right, understood. Is there a rough timeline on the horizon for v4? 

Many thanks.

**Reply by marina-mosti** [reply]
> Right, understood. Is there a rough timeline on the horizon for v4? 
> 
> 
> 
> Many thanks.

Not yet. It's only us 2 maintaining the lib at the moment and there's a lot of things we want/need to test and consider so it'll take quite a bit of work. 
It really all depends on how much work we have and if burnout permits :)

**Reply by jhrr** [reply]
I understand that! And I appreciate the library and all the hard work.
